cdap-ui/app/cdap/components/FileBrowser/index.js

```diff
@@ -464,7 +478,11 @@ export default class FileBrowser extends Component {
 
         <div className="content-body clearfix">
           {displayContent.map((content) => {
-            return this.renderRow(content);
+            let isRowSelected = false;
+            if (inWorkspaceDirectory && content.name === this.getFileName()) {
```

Comment:
Can we do this check directly in the `renderRowContent`? Seems like ```this.state.path === this.getFilePath() && content.name === this.getFileName()``` should be available in the specific function. 

Avoids passing this prop down multiple levels.

Reply:
You can do the full check there, but my concern was that perform the path check once per row was excessive. If you think it's not a concern, I can make the change.

Reply:
I think that should be ok. Its a constant time operation (1 check).